### ISSUE TITLE
  How do I identify or visualize the av...

### ISSUE THREAD

**Issue by wormbasehelp** [WormBase/website#6049]
*Submitted from the feedback form on the WormBase website.*

<p>&nbsp;&nbsp;How do I identify or visualize the available duplications available for a particular region of the genome in the Genome Browser?<br/>Thank you for your help!</p>

**Reported by:** Dian* (dlib***************)
**Submitted from:** <a target="_blank" href="http://www.wormbase.org//tools/support">/tools/support</a>
**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/604.3.5 (KHTML, like Gecko) Version/11.0.1 Safari/604.3.5

**Comment by chris-grove** [comment]
Hello Diana,


 My colleagues can correct me if I'm wrong, but I don't think either of our Genome Browsers (GBrowse or JBrowse) have a track, or other mechanism, to view duplications or other rearrangements on the genome. I think that currently the best place to visualize these is on the Genetic Map viewer:


http://www.wormbase.org/tools/gmap


You can search for a particular chromosome, gene, or rearrangement to focus on.


Best,


Chris Grove

WormBase

**Comment by chris-grove** [comment]
@scottcain 
Maybe I should leave this open as a feature request? Is this something that JBrowse could do?

**Comment by scottcain** [comment]
@chris-grove I was thinking about this.  On a very simple level, you could imagine a blast service, where you select a region and JBrowse then displays the long "self matches" where they aren't in the same location.  Of course, on a more "curated" level, I'm guessing this information already exists.  @khowe is there a reason we don't have a track like this?

**Comment by Paul-Davis** [comment]
Hi Diana,

There is actually a set of segmental duplications calculated by 
OrthoCluster that can be viewed in the both of our genome browsers for 
C. elegans.

http://www.wormbase.org/tools/genome/jbrowse-simple/?tracks=Curated_Genes%2CYACs_Fosmids_Cosmids%2CSegmental%20duplication&loc=III%3A9044186..9051990&data=data%2Fc_elegans_PRJNA13758&highlight=

http://www.wormbase.org/species/c_elegans/feature/WBsf039960#012374--10

Unfortunately you cannot currently click in the feature in gbrowse and 
get to the summary pages but the ID is searchable and can be cut and 
pasted from the popup window.

These were taken directly from a 3rd party submitter so we do not have 
an accurate assessment of the accuracy or completeness but the details 
are published in this paper: 
http://www.wormbase.org/resources/paper/WBPaper00034747#05--10

I hope this helps.

Paul

On 14/12/2017 16:21, Grove, Christian A. (Chris) wrote:
>
> Hello Diana,
>
>
>  My colleagues can correct me if I'm wrong, but I don't think either 
> of our Genome Browsers (GBrowse or JBrowse) have a track, or other 
> mechanism, to view duplications or other rearrangements on the genome. 
> I think that currently the best place to visualize these is on the 
> Genetic Map viewer:
>
>
> http://www.wormbase.org/tools/gmap
>
>
> You can search for a particular chromosome, gene, or rearrangement to 
> focus on.
>
>
> Best,
>
>
> Chris Grove
>
> WormBase
>
> ------------------------------------------------------------------------
> *From:* no-reply@wormbase.org <no-reply@wormbase.org>
> *Sent:* Thursday, December 14, 2017 12:42:24 AM
> *To:* help@wormbase.org
> *Cc:* dlibuda@uoregon.edu
> *Subject:* [wormbase-help] How Do I ...?: How do I identify or 
> visualize the av... (Diana)
> How Do I ...?
>
>   How do I identify or visualize the available duplications available 
> for a particular region of the genome in the Genome Browser?
> Thank you for your help!
>
> Submitted From:
>   support /tools/support <http://www.wormbase.org//tools/support>
>
> This issue has been assigned ticket number: #6049. Follow progress on 
> our issue tracker <https://github.com/WormBase/website/issues/6049>.
>
> Reported by: Diana (dlibuda@uoregon.edu)
> 14 Dec 17 08:42
>
> Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/604.3.5 
> (KHTML, like Gecko) Version/11.0.1 Safari/604.3.5
>
> This question was reported via the web interface at wormbase.org. You 
> can also email us <mailto:help@wormbase.org> directly.
>
> -- 
> -- 
> You received this message because you are subscribed to the "WormBase: 
> help" group.
> To post to this group, send email to help@wormbase.org
> To unsubscribe from this group, send email to
> help+unsubscribe@wormbase.org
> For more options, visit this group at
> http://groups.google.com/a/wormbase.org/group/help?hl=en
>
> ---
> You received this message because you are subscribed to the Google 
> Groups "WormBase: help" group.
> To unsubscribe from this group and stop receiving emails from it, send 
> an email to help+unsubscribe@wormbase.org 
> <mailto:help+unsubscribe@wormbase.org>.
> -- 
> -- 
> You received this message because you are subscribed to the "WormBase: 
> help" group.
> To post to this group, send email to help@wormbase.org
> To unsubscribe from this group, send email to
> help+unsubscribe@wormbase.org
> For more options, visit this group at
> http://groups.google.com/a/wormbase.org/group/help?hl=en
>

-- 
===========================================
WormBase
European Bioinformatics Institute (EMBL-EBI)
European Molecular Biology Laboratory
Wellcome Trust Genome Campus
Hinxton
Cambridge CB10 1SD
United Kingdom
===========================================

**Comment by Paul-Davis** [comment]
@scottcain Could the linking from the above track get changed to look for a Feature class object and not do a "Sequences" search pls?

**Comment by chris-grove** [comment]
@Paul-Davis 
I would need to ask Diana to clarify, but I was under the impression she was asking about visualization of ?Rearrangement objects in the genome browser, not segmental duplication ?Feature objects.

**Comment by khowe** [comment]
@chris-grove Looking at the model for ?Rearrangement, their locations are defined genetically rather than genomically. If we wanted to show them on the genome browser, we would need to create a new step in the build that creates "fake" genomic positions for each one, based on their genetic position. 

But even if we did that, there are different types of rearrangement, and some of them don't visualise cleanly as a browser track (e.g. translocation). But I guess the track could simply mark up regions that are "rearranged" in some way, with a pop-up giving details of the actual rearrangement.

**Comment by Paul-Davis** [comment]
ah right, I didn't appreciate she might be after the rearrangement class. Would be good to get the feature linking fixed anyway :)

**Comment by chris-grove** [comment]
@khowe 
Yeah, I understand that that might be a challenge. Maybe we could implement "fuzzy" boundaries somehow? I guess I'm just thinking of somehow replicating the Genetic Map viewer but in the context of the genome browser.
@Paul-Davis 
Yes, I agree we need to fix the feature linking. I'll create another ticket with your request, just to keep tabs on both issues.

**Comment by Paul-Davis** [comment]
ok, thanks @chris-grove 👍

**Comment by chris-grove** [comment]
OK @Paul-Davis 
The new issue is #6054

**Comment by tharris** [comment]
@khowe and @chris-grove: we already interpolate the genetic map into physical coordinates — the “genetic limits” track for genetic loci. Like @khowe mentioned it wouldn’t work for major rearrangements, but I’m sure it could be expanded to handle some cases.
